File: solutionFiles/LongestSubstringWithoutRepeatingCharacters.js
function lengthOfLongestSubstring(s) {
    let charSet = new Set();
    let left = 0;
    let maxLength = 0;

    for (let right = 0; right < s.length; right++) {
        while (charSet.has(s[right])) {
            charSet.delete(s[left]);
            left++;
        }
        charSet.add(s[right]);
        maxLength = Math.max(maxLength, right - left + 1);
    }

    return maxLength;
}

function main() {
    if (process.argv.length !== 3) {
        console.log("Please provide a single JSON string argument.");
        return;
    }

    let inputJson = process.argv[2];
    let inputData;
    try {
        inputData = JSON.parse(inputJson);
    } catch (e) {
        console.log("Invalid JSON input.");
        return;
    }

    let s = inputData.s || "";
    let result = lengthOfLongestSubstring(s);
    let output = { result: result };
    console.log(JSON.stringify(output));
}

main();

File: solutionFiles/LongestSubstringWithoutRepeatingCharacters.py
import sys
import json

def length_of_longest_substring(s: str) -> int:
    char_set = set()
    left = 0
    max_length = 0

    for right in range(len(s)):
        while s[right] in char_set:
            char_set.remove(s[left])
            left += 1
        char_set.add(s[right])
        max_length = max(max_length, right - left + 1)

    return max_length

def main():
    if len(sys.argv) != 2:
        print("Please provide a single JSON string argument.")
        return

    input_json = sys.argv[1]
    try:
        input_data = json.loads(input_json)
        s = input_data.get("s", "")
    except json.JSONDecodeError:
        print("Invalid JSON input.")
        return

    result = length_of_longest_substring(s)
    output = {"result": result}
    print(json.dumps(output))

if __name__ == "__main__":
    main()
